New: Added new lines of code.

HKilaji opened the pull request:
-Member functions set_snakeTail(),move_snakeHead() added to class Snake in file Snake.h

-Member functions get_gameover(),increase_score() and get_score() added to class Game in file Game.h.

-main.cpp file adjusted according to previous changes.

HKilaji committed:
feat: Adding new program files.

-Frame.h is added. Contains class Frame, contains member function for size of board, contains draw function to draw frame around the board.

-main.cpp is added. Contains object for class Frame, and a draw function inside a loop.

HKilaji committed:
code is fixed according to the last review comment:

- New parts are added to the code. Class Food, class Snake and class game are added.

- Changes done to the class Frame to address to the review comment and according to the new addition to code.

- Code is still not in working condition.there is segmentation fault error in Game::draw(). will work on that on next update.

HKilaji committed:
Fix: New lines of codes are added and fix pervious issues.

-Changes done to remove and add blank lines. Data members are set to private.

-New classes Food, Frame and Snake are created in corresponding files.

-Classes Seperation to Corresponding header and Cpp Files should be addressed in next update.

-Makefile is created. Changes will be done to makefile after Class file Seperation.

HKilaji committed:
Delete play

HKilaji committed:
Delete a.out

HKilaji committed:
Update Frame.h

HKilaji committed:
New: Added new lines of code.

-Member functions set_snakeTail(),move_snakeHead() added to class Snake in file Snake.h

-Member functions get_gameover(),increase_score() and get_score() added to class Game in file Game.h.

-main.cpp file adjusted according to previous changes.

HKilaji committed:
Merge remote-tracking branch 'refs/remotes/origin/main' into main

HKilaji committed:
resolved merge conflicts

johansaji commented on Frame.h:
```diff
@@ -0,0 +1,38 @@
+		}
+		for( int itr_width = 0 ; itr_width < width ; itr_width ++){
+			screen[itr_width][itr_height] = symbol_wall ;
+<<<<<<< Updated upstream
+		}
+=======
+		}	
+>>>>>>> Stashed changes
```
hope the code is compilable?

HKilaji replied:
yes It is compilable.

johansaji commented on Snake.h:
```diff
@@ -0,0 +1,76 @@
+#include <stdio.h>
+
+class Game;
```
why cant you include the header file?

HKilaji replied:
which one, Game.h? it is not showing any error so I leave that as it is. will check up if you want me to include it.

johansaji commented on Game.h:
```diff
@@ -0,0 +1,46 @@
+	int score = 0;
+public:
+	Game(){
+		screen = new char * [height];
```
where are you freeing this memory allocated?

HKilaji replied:
I am planning to do it in destructor of class Game. but will check if it allows me to because I need added code for input and logic of game. I don't want to free it before necessary.  Will definitely write the code to free it, but a bit later.

HKilaji committed:
New: Added new line of code and header and cpp files seperation done

- Created new cpp files corresponding to header files. modified Makefile and all Header files according to this create.

- Changes done in set_snakeTail() and added move_tail() and get_Snakelength() in class Snake.

- Created a destructor to free memory of 2d array screen in class Game.

- Removed unexpected lines of code causing issue >>>>>upstream.

- Removed class Game; forward declaration from classes Snake, Food, Frame.
